### PR TITLE
Use the default pipe options for backpressure

### PR DESCRIPTION
**Pull request by davidfowl**
- This controls memory so clients aren't easily overwhelmed. With the changes made to Pipe to no longer throw if the pause threshold is exceeded makes this work well.

Contributes to #17797

Here's the sample from https://github.com/dotnet/aspnetcore/issues/20654

Before:

image

After:

image

### CODE REVIEW

File: src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/ServerSentEventsTransport.cs [hunk 1]
```diff
@@ -133,8 +133,7 @@ private async Task ProcessEventStream(HttpResponseMessage response, Cancellation
             using (response)
             using (var stream = await response.Content.ReadAsStreamAsync())
             {
-                var options = new PipeOptions(pauseWriterThreshold: 0, resumeWriterThreshold: 0);
-                var reader = PipeReaderFactory.CreateFromStream(options, stream, cancellationToken);
+                var reader = PipeReader.Create(stream);
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/ServerSentEventsTransport.cs:136]: Missing the cancellation token

**Comment by BrennanConroy** [comment]
Are we not going to add the option in this PR?

**Comment by davidfowl** [comment]
> Are we not going to add the option in this PR?

I chose not to but I can. That's why I wrote contributes to vs fixes. The option isn't as important anymore because it's no longer a hard limit in pipes.

**Comment by BrennanConroy** [comment]
Does this affect throughput significantly?

**Comment by davidfowl** [comment]
> Does this affect throughput significantly?

In this test it improves throughput since the GC is basically bringing the application to its knees. I haven't run any benchmarks though.

**Comment by davidfowl** [comment]
# Before

## Counters:
```
CPU Usage (%):                 58
Working Set (MB):              75
GC Heap Size (MB):             13
Gen 0 GC (#/s):                36
Gen 1 GC (#/s):                4
Gen 2 GC (#/s):                0
Time in GC (%):                12
Gen 0 Size (B):                180,784
Gen 1 Size (B):                1,154,376
Gen 2 Size (B):                13,742,736
LOH Size (B):                  48,416
Allocation Rate (B/sec):       272,120,129
# of Assemblies Loaded:        66
Exceptions (#/s):              0
ThreadPool Threads Count:      43
Lock Contention (#/s):         959
ThreadPool Queue Length:       65
ThreadPool Items (#/s):        244,659
```

## Benchmark:

```
Max RPS:                       593,436
Requests:                      8,918,162
Mean latency (ms):             9
```

# After

## Counters:

```
CPU Usage (%):                 63
Working Set (MB):              68
GC Heap Size (MB):             13
Gen 0 GC (#/s):                32
Gen 1 GC (#/s):                3
Gen 2 GC (#/s):                0
Time in GC (%):                10
Gen 0 Size (B):                274,304
Gen 1 Size (B):                1,048,648
Gen 2 Size (B):                12,286,616
LOH Size (B):                  48,416
Allocation Rate (B/sec):       239,139,013
# of Assemblies Loaded:        65
Exceptions (#/s):              0
ThreadPool Threads Count:      56
Lock Contention (#/s):         926
ThreadPool Queue Length:       79
ThreadPool Items (#/s):        290,194
```

## Benchmark:

```
Max RPS:                       561,548
Requests:                      8,444,560
Mean latency (ms):             5
```

Slightly slower on our broadcast benchmark same on echo.

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**